src/evaluate.jl

```diff
@@ -96,43 +96,42 @@ end
 Evaluate a `HomogeneousPolynomial` polynomial using Horner's rule (hand coded)
 at `vals`.
 """
-function evaluate{T<:Number,S<:Union{Real,Complex}}(a::HomogeneousPolynomial{T},
-    vals::Array{S,1} )
-
-    numVars = get_numvars()
-    @assert length(vals) == numVars
+# function evaluate{T<:Number}(a::HomogeneousPolynomial{T}, vals::Array{T,1} )
+function evaluate{T<:Number,S<:NumberNotSeriesN}(a::HomogeneousPolynomial{T},
+        vals::Array{S,1} )
+    @assert length(vals) == get_numvars()
     R = promote_type(T,S)
+    numVars = get_numvars()
     suma = convert(TaylorN{R}, a)
```

Comment:
Type stability; if you simply use assignment (`suma = a`) it may change type inside the loop.